# HG changeset patch
# User Rik <rik@octave.org>
# Date 1516131987 28800
#      Tue Jan 16 11:46:27 2018 -0800
# Branch stable
# Node ID 60d6914d801be99736991e378749b04322ba6a17
# Parent  e2cf81bcdf01a85a84a3c9ea65212c1a74bd5adb
Don't create partially invalid graphic objects (bug #52904).

* graphics.cc (make_graphics_object): Wrap xset() call within try/catch block
in case additional arguments given to primitive object cause a failure.  If
error caught, delete the partially formed object before issuing an error.
Also, correct function identified in error message to be "__go_%s__".

diff --git a/libinterp/corefcn/graphics.cc b/libinterp/corefcn/graphics.cc
--- a/libinterp/corefcn/graphics.cc
+++ b/libinterp/corefcn/graphics.cc
@@ -10421,23 +10421,33 @@ make_graphics_object (const std::string&
   try
     {
       h = gh_manager::make_graphics_handle (go_name, parent,
                                             integer_figure_handle,
                                             false, false);
     }
   catch (octave::execution_exception& e)
     {
-      error (e, "__go%s__: unable to create graphics handle",
+      error (e, "__go_%s__: unable to create graphics handle",
              go_name.c_str ());
     }
 
   adopt (parent, h);
 
-  xset (h, xargs);
+  try
+    {
+      xset (h, xargs);
+    }
+  catch (octave::execution_exception& e)
+    {
+      delete_graphics_object (h);
+      error (e, "__go_%s__: unable to create graphics handle",
+             go_name.c_str ());
+    }
+
   xcreatefcn (h);
   xinitialize (h);
 
   retval = h.value ();
 
   Vdrawnow_requested = true;
 
   return retval;
